### PR TITLE
1.19.3

### PR DESCRIPTION
**Pull request by BonbonGit**
Tried to port to 1.19.3

I have honestly no idea what I am doing and just tried to find equivalent classes/methods in the new versions of Fabric and Minecraft.

Which results in missing textures for pipes for now, but otherwise the mod seems to work.

Also you may want to merge it in another branch since I don't think build for 1.19.3 can be made compatible with older versions.

### CODE REVIEW

File: src/client/java/dev/technici4n/moderndynamics/client/model/PipeUnbakedModel.java [hunk 1]
```diff
@@ -64,23 +60,21 @@ public Collection<ResourceLocation> getDependencies() {
...
+        unbakedModelGetter.apply(AttachmentsUnbakedModel.ID).resolveParents(unbakedModelGetter);
     }
 
-    @Nullable
     @Override
-    public BakedModel bake(ModelBakery modelBakery, Function<Material, TextureAtlasSprite> spriteGetter, ModelState transform,
+    public BakedModel bake(ModelBaker modelBaker, Function<Material, TextureAtlasSprite> spriteGetter, ModelState transform,
             ResourceLocation location) {
         return new PipeBakedModel(
+                // Load transform from the vanilla block model
```
Comment (Technici4n) [inline src/client/java/dev/technici4n/moderndynamics/client/model/PipeUnbakedModel.java:73]: Why?

**Review by Technici4n — CHANGES_REQUESTED** [review]
Looks like a good start, thank you. For the creative tab: tabs are now directly responsible for their contents. For MD the registration order is fine: we can simply add all Item registry entries that live in the MD namespace.

**Comment by Technici4n** [comment]
Oh, and you need to run `gradlew runData` to generate some models, the recipes, the attachment upgrades, etc...

**Comment by BonbonGit** [comment]
> Looks like a good start, thank you. For the creative tab: tabs are now directly responsible for their contents. For MD the registration order is fine: we can simply add all Item registry entries that live in the MD namespace.

Added items to the creative tab. Not sure that's how you meant to do it but it works.

> Oh, and you need to run `gradlew runData` to generate some models, the recipes, the attachment upgrades, etc...

I get recipes, upgrades, etc. now and from what I understand models load too but not textures:

image

But I don't undestand what is supposed to load the textures so I'm stuck here.

File: src/client/java/dev/technici4n/moderndynamics/client/model/MdModels.java [hunk 2]
```diff
@@ -40,7 +39,8 @@ public class MdModels {
     };
 
     public static ModelState preRotated(BlockModelRotation rotation, float preAngle) {
-        Transformation preRotation = new Transformation(null, new Quaternion(new Vector3f(0, 0, 1), preAngle, true), null, null);
+
+        Transformation preRotation = new Transformation(null, new Quaternionf(0, 0, 1, 0), null, null);
```
Comment (BonbonGit) [inline src/client/java/dev/technici4n/moderndynamics/client/model/MdModels.java:43]: Sorry I'm clueless about geometry so I have no idea what the line previously there or what I replaced it with does, but it works somehow.

**Comment by Technici4n** [comment]
Thanks a lot, I'll have a look in a few days. I have another feature in the pipeline that I want done before I merge this.

**Comment by ewpratten** [comment]
Hey, just hopping in to say that I have been running this branch on a small server for a few days, and everything works great!

**Comment by Technici4n** [comment]
Good to know. However there won't be any official 1.19.3 release, and I'm waiting on 1.20 instead.

**Pull request closed without merging**